backend/src/test/kotlin/org/loculus/backend/controller/submission/GetReleasedDataEndpointTest.kt

```diff
@@ -118,6 +125,35 @@ class GetReleasedDataEndpointTest(
         }
     }
 
+    @Test
+    fun `GIVEN header etag gt last db update THEN respond with 304, ELSE respond with data and etag`() {
+        convenienceClient.prepareDefaultSequenceEntriesToApprovedForRelease()
+
+        val response = submissionControllerClient.getReleasedData()
+        val responseBody = response.expectNdjsonAndGetContent<ProcessedData<GeneticSequence>>()
+
+        val mvcResult: MvcResult = response.andReturn()
+
+        val initialEtag = mvcResult.response.getHeader(ETAG)
+        assertThat(responseBody.size, `is`(NUMBER_OF_SEQUENCES))
+
+        val responseNoNewData = submissionControllerClient.getReleasedData(
+            ifNoneMatch = initialEtag,
+        )
+        responseNoNewData.andExpect(status().isNotModified)
+            .andExpect(header().doesNotExist(ETAG))
+
+        prepareRevokedAndRevocationAndRevisedVersions()
+
+        val responseAfterMoreDataAdded = submissionControllerClient.getReleasedData(
+            ifNoneMatch = initialEtag,
+        )
+
+        responseAfterMoreDataAdded.andExpect(status().isOk)
+            .andExpect(header().string(ETAG, notNullValue()))
+            .andExpect(header().string(ETAG, greaterThan(initialEtag)))
```

Comment:
Let's maybe add an assertion that the response body contains something?

Reply:
done :-)

Reply:
Done :-)